backend/Dockerfile

```diff
@@ -12,47 +12,79 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Build arguments
-ARG SOURCE_CODE=.
+# 1. Build api server application
+FROM golang:1.21.7-bookworm as builder
+RUN apt-get update && apt-get install -y cmake clang musl-dev openssl
+WORKDIR /go/src/github.com/kubeflow/pipelines
 
-FROM registry.access.redhat.com/ubi8/go-toolset:1.21 as builder
```

Comment:
Could you clarify on why we're removing `registry.access.redhat.com/ubi8/go-toolset:1.21` from here and other dockerfiles?

Reply:
We want to leave the upstream Dockerfiles unchanged to avoid merge conflicts during rebase. Instead, we relocated the DSP Dockerfiles to the odh_images folder